subprojects/docs/src/docs/userguide/dep-man/01-core-dependency-management/dependency_verification.adoc

```diff
@@ -0,0 +1,849 @@
+
+Alternatively, you might want to enable verification on CI servers but not on local machines.
+
+Gradle actually provides 3 different verification modes:
+
+- `strict`, which is the default.
+Verification fails _as early as possible_, in order to avoid the use of compromised dependencies during the build.
+- `lenient`, which will run the build even if there are verification failures.
+The verification errors will be displayed during the build without causing a build failure.
+- `off` when verification is totally ignored.
+
+All those modes can be activated on the CLI using the `--dependency-verification` flag, for example:
+
+
+----
+./gradlew --dependency-verification lenient build
+----
+
+Alternatively, you can set the `org.gradle.dependency.verification` system property, either on the CLI:
+
+
+----
+./gradlew -Dorg.gradle.dependency.verification=lenient build
+----
+
+or in a `gradle.properties` file:
+
+
+----
+systemProp.org.gradle.dependency.verification=lenient
+----
+
+[[sec:trusting-artifacts]]
+=== Trusting some particular artifacts
+
+You might want to trust some artifacts more than others.
+For example, it's legitimate to think that artifacts produced in your company and found in your internal repository only are safe, but you want to check every external component.
+
+[NOTE]
+====
+This is a typical _company policy_.
+In practice, **nothing** prevents your internal repository from being compromised, so it's a good idea to check your internal artifacts too!
+====
+
+For this purpose, Gradle offers a way to automatically trust some artifacts.
+You can trust all artifacts in a group by adding this to your configuration:
+
+[source,xml]
+----
+<?xml version="1.0" encoding="UTF-8"?>
+<verification-metadata>
+   <configuration>
+      <trusted-artifacts>
+         <trust group="com.mycompany"/>
+      </trusted-artifacts>
+   </configuration>
+</verification-metadata>
+----
+
+This means that all components which group is `com.mycompany` will automatically be trusted.
+Trusted means that Gradle will not perform any verification whatsoever.
+
+The `trust` element accepts those attributes:
+
+- `group`, the group of the artifact to trust
+- `name`, the name of the artifact to trust
+- `version`, the version of the artifact to trust
+- `file`, the name of the artifact _file_ to trust
+- `regex`, a boolean saying if the `group`, `name`, `version` and `file` attributes need to be interpreted as regular expressions (defaults to `false`)
+
+In the example above it means that the trusted artifacts would be artifacts in `com.mycompany` but not `com.mycompany.other`.
+To trust all artifacts in `com.mycompany` and all subgroups, you can use:
+
+[source,xml]
+----
+<?xml version="1.0" encoding="UTF-8"?>
+<verification-metadata>
+   <configuration>
+      <trusted-artifacts>
+         <trust group="^com[.]mycompany($|([.].*))" regex="true"/>
```

Comment:
Does `regex` use `Matcher#matches`, `Matcher#lookingAt`, or `Matcher#find` method? (see https://docs.oracle.com/javase/7/docs/api/java/util/regex/Matcher.html )

If `matches` (entire region) or `lookingAt` (prefix search) is used (I would expect `matches`), then `^` is not required.
If `find` is used, then it makes sense to add a recommendation to prefer having `^`

Reply:
it uses `matches`: https://github.com/gradle/gradle/blob/06f812f4a1160a40785fc11720667598b8ef8197/subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/verification/verifier/DependencyVerificationConfiguration.java#L119